skale/transactions/result.py

```diff
@@ -0,0 +1,99 @@
+#   -*- coding: utf-8 -*-
+#
+#   This file is part of SKALE.py
+#
+#   Copyright (C) 2019-Present SKALE Labs
+#
+#   SKALE.py is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU Affero General Public License as published by
+#   the Free Software Foundation, either version 3 of the License, or
+#   (at your option) any later version.
+#
+#   SKALE.py is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with SKALE.py.  If not, see <https://www.gnu.org/licenses/>.
+
+
+SUCCESS_STATUS = 1
+
+
+class TransactionError(Exception):
+    pass
+
+
+class DryRunFailedError(TransactionError):
+    pass
+
+
+class InsufficientBalanceError(TransactionError):
+    pass
+
+
+class TransactionFailedError(TransactionError):
+    pass
+
+
+def check_balance(balance: int, gas_price: int, gas_limit: int) -> dict:
+    tx_cost = gas_price * gas_limit
+    if balance < tx_cost:
+        status = 0
+        msg = f'Transaction requires {tx_cost}. Wallet has {balance} wei'
+    else:
+        status = 1
+        msg = 'ok'
+    return {'status': status, 'msg': msg}
+
+
+def is_success(result: dict) -> bool:
+    return result.get('status') == SUCCESS_STATUS
+
+
+def is_success_or_not_performed(result: dict) -> bool:
+    return result is None or is_success(result)
+
+
+class TxRes:
```

Comment:
Are we going to keep this name here?

Reply:
Didn't get the point. A little more explanation needed